src/workerd/jsg/resource.h

```diff
@@ -531,6 +624,75 @@ struct SetterCallback<TypeWrapper, methodName,
   }
 };
 
+template <typename TypeWrapper, const char* methodName,
+          typename Method, Method method, bool isContext>
+struct PropertySetterCallback;
+
+template <typename TypeWrapper, const char* methodName, typename T, typename Arg,
+          void (T::*method)(Arg), bool isContext>
+struct PropertySetterCallback<TypeWrapper, methodName, void (T::*)(Arg), method, isContext> {
+  static void callback(const v8::FunctionCallbackInfo<v8::Value>& info) {
+    liftKj(info, [&]() {
+      auto isolate = info.GetIsolate();
+      auto context = isolate->GetCurrentContext();
+      auto obj = info.This();
+      auto& wrapper = TypeWrapper::from(isolate);
+      // V8 no longer supports AccessorSignature, so we must manually verify `this`'s type.
+      if (!isContext && !wrapper.template getTemplate(isolate, (T*)nullptr)->HasInstance(obj)) {
+        throwTypeError(isolate, "Illegal invocation");
+      }
+      auto& self = extractInternalPointer<T, isContext>(context, obj);
+      (self.*method)(wrapper.template unwrap<Arg>(context, info[0],
+          TypeErrorContext::setterArgument(typeid(T), methodName)));
+    });
+  }
+};
+
+// Specialization for methods that take `Lock&` as their first parameter.
+template <typename TypeWrapper, const char* methodName, typename T, typename Arg,
+          void (T::*method)(Lock&, Arg), bool isContext>
+struct PropertySetterCallback<TypeWrapper, methodName,
+                      void (T::*)(Lock&, Arg), method, isContext> {
+  static void callback(const v8::FunctionCallbackInfo<v8::Value>& info) {
+    liftKj(info, [&]() {
+      auto isolate = info.GetIsolate();
+      auto context = isolate->GetCurrentContext();
+      auto obj = info.This();
+      auto& wrapper = TypeWrapper::from(isolate);
+      // V8 no longer supports AccessorSignature, so we must manually verify `this`'s type.
+      if (!isContext && !wrapper.template getTemplate(isolate, (T*)nullptr)->HasInstance(obj)) {
+        throwTypeError(isolate, "Illegal invocation");
+      }
+      auto& self = extractInternalPointer<T, isContext>(context, obj);
+      (self.*method)(Lock::from(isolate), wrapper.template unwrap<Arg>(context, info[0],
+          TypeErrorContext::setterArgument(typeid(T), methodName)));
+    });
+  }
+};
+
+// Specialization for methods that take `const v8::PropertyCallbackInfo<void>&` as their
+// first parameter.
+template <typename TypeWrapper, const char* methodName, typename T, typename Arg,
```

Comment:
Note for reviewers: likewise here... we don't have anything that's actually using this specialization. We can simplify by removing this an the `v8::PropertyCallbackInfo` variation